src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -207,6 +244,12 @@ void DlgPrefColors::slotApply() {
                         m_colorPaletteSettings.getTrackColorPalette()));
     }
 
+    if (!bKeyColorPaletteFound) {
+        m_colorPaletteSettings.setKeyColorPalette(
+                m_colorPaletteSettings.getColorPalette(keyColorPaletteName,
+                        m_colorPaletteSettings.getKeyColorPalette()));
+    }
```

Comment:
I'm not happy with the amount of string roundtrips here....

Reply:
This is a string based lookup, because the user is able to specify custom palettes.
There might be a workaround for build in pallets, but since this is not a hot path I like to avoid special case handling here. Or do you have something different in mind?

Reply:
I see. disregard my comment.